src/DoctrineObject.php

```diff
@@ -314,6 +316,11 @@ public function hydrateValue(string $name, $value, ?array $data = null)
             return null;
         }
 
+        // BackedEnum is available from PHP 8.1 on
+        if (interface_exists(BackedEnum::class) && $value instanceof BackedEnum) {
```

Comment:
`interface_exists(BackedEnum::class, false)` should be used, to prevent autoloading here.

Reply:
I see: https://3v4l.org/ru7sL. I was expecting this to throw a warning on older versions, which obviously it doesn't.

Reply:
Fixed it

Reply:
Yeah, `instanceof` has no side-effects, luckily! :D